aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/OrchestratorTest.java

```diff
@@ -100,8 +113,17 @@ public void itemsPlacedInStorageArePublishedToNetwork() throws AmplifyException
             );
 
         // Arrange: orchestrator is running
-        orchestrator.start();
+        orchestrator.start().subscribe();
+
+        // Try to start it in a new thread.
+        new Thread(() -> orchestrator.start().subscribe()).start();
+        // Try to start it again on a current thread.
+        orchestrator.start().subscribe();
```

Comment:
I understand that you are testing the semaphore, but can we move this to a separate test case? Someone without context checking out this test might get confused why this step is necessary in testing that the "items placed in storage are published to network".

Reply:
That's a good point. I'm breaking this out.